cv_next/app/cv/[cvId]/components/commentSection/comment.tsx

```diff
@@ -0,0 +1,114 @@
+"use client";
+import { deleteComment } from "@/app/actions/comments/deleteComment";
+import { setResolved } from "@/app/actions/comments/setResolved";
+import { upvoteComment } from "@/app/actions/comments/setLike";
+import { mutate } from "swr";
+
+export default function Comment({
+  comment,
+  userId,
+  childDepth = 0,
+}: {
+  comment: CommentModel;
+  userId: string;
+  childDepth?: number;
+}) {
+  const date = new Date(comment.last_update);
+  const childOrParentStyling = childDepth
+    ? `p-3 ml-${6 / childDepth} lg:ml-${12 / childDepth} border-t border-gray-400 dark:border-gray-600`
+    : "p-6 mb-3 border-b border-gray-200 rounded-lg";
+
+  const deleteCommentAction = async () => {
+    await deleteComment(comment.id).finally(() => {
+      mutate(comment.document_id);
+    });
+  };
+
+  const setResolvedCommentAction = async (resolved: boolean) => {
+    await setResolved(comment.id, resolved).finally(() => {
+      mutate(comment.document_id);
+    });
+  };
+
+  const setLikedCommentAction = async (liked: boolean) => {
+    await upvoteComment(comment.id, liked, userId).finally(() => {
+      mutate(comment.document_id);
+    });
+  };
+
+  const commenter = JSON.parse(JSON.stringify(comment.user_id));
+  const userName =
+    commenter.username && commenter.username.length > 0
+      ? commenter.username
+      : commenter.full_name;
+
+  const voteingSection =
+    comment.upvotes && comment.upvotes.includes(userId) ? (
+      <button
+        className="text-gray-500"
+        onClick={() => setLikedCommentAction(false)}
+      >
+        UnLike {comment.upvotes?.length || 0}
+      </button>
+    ) : (
+      <button
+        className="text-gray-500"
+        onClick={() => setLikedCommentAction(true)}
+      >
+        Like {comment.upvotes?.length || 0}
+      </button>
+    );
```

Comment:
It seems here that the conditional is only on the boolean value of `setLikedCommentAction`?
If so, it'd probably be better to replace the boolean value with the boolean check.
E.g:
```suggestion
  const userVoted = comment.upvotes.includes(userId);
  const voteingSection =
    comment.upvotes &&
     (
      <button
        className="text-gray-500"
        onClick={() => setLikedCommentAction(userVoted)}
      >
        {userVoted ? 'Unlike' : 'Like'} {comment.upvotes?.length || 0}
      </button>
    );
```

BTW I think using like icon with enable/disable states is both easier and more UX-friendly, for the like label

Reply:
The text and labels are updated in the branch Adam is working on